Source/CombatExtended/CombatExtended/StatWorkers/StatWorker_Caliber.cs

```diff
@@ -28,23 +28,27 @@ private Thing Gun(StatRequest req)
 
         public override bool ShouldShowFor(StatRequest req)
         {
-            return base.ShouldShowFor(req) && 
-                (Controller.settings.EnableAmmoSystem
-                    ? GunDef(req)?.GetCompProperties<CompProperties_AmmoUser>()?.ammoSet != null
-                    : (GunDef(req)?.Verbs?.Any(x => x.defaultProjectile != null) ?? false));
+            if (!base.ShouldShowFor(req)) return false;
+
+            AmmoSetDef ammoSet = GunDef(req)?.GetCompProperties<CompProperties_AmmoUser>()?.ammoSet;
+            if (Controller.settings.EnableAmmoSystem || (ammoSet != null && ammoSet.isMortarAmmoSet))
```

Comment:
This check appears 4 or 5 times, can it be refactored? (It also looks very similar to the check in `CompAmmoUser#UseAmmo`)